### ISSUE TITLE
[Bug]: WPF app (.NET6) crash because of stackoverflow (i guess) in the CoreWCF Service

### ISSUE THREAD

**Issue by Montago** [CoreWCF/CoreWCF#1267]
### Duplicate ?

- [X] I have searched issues/discussions and did not find other issues/discussions reporting this bug.

### Product version

1.4.1

### Describe expected behavior

An error message that can point me in the right direction to solve an issue like this. 

### Describe actual behavior

When calling a method that by accident caused stackoverflow on the Service, the WPF Application completely crashed silently without any error message. 

### Which binding

BasicHttp

### security

Transport

### Which .NET version

.NET 6

### Which os platform

Windows

### Code snippet used to reproduce the issue

```c#
Basically i had a method on the service that called itself: 

public bool RecusiveCall()
{
    return RecusiveCall()
}

which makes the Service explode / crash in such a bombastic manor that the WPF application that called the method, also explode - but silently without any errors at all :(
```

**Comment by mconnew** [comment]
According to [documentation for StackOverflowException](https://learn.microsoft.com/en-us/dotnet/api/system.stackoverflowexception?view=net-7.0&redirectedfrom=MSDN#remarks), this is expected behavior:

> Starting with the .NET Framework 2.0, you can't catch a StackOverflowException object with a try/catch block, and the corresponding process is terminated by default. Consequently, you should write your code to detect and prevent a stack overflow.

There's nothing WCF, CoreWCF, or WPF can do about it. The reason is that the process is effectively in a corrupt state. The only time a StackOverflowException can be caught is when user code throws it, but if the runtime throws it, there's nothing you can do.

Closing as this is an intentional design of the runtime and not affected by anything WCF/CoreWCF is doing.
